src/Engine/AttackList.cpp

```diff
@@ -0,0 +1,17 @@
+#include "Engine/AttackList.h"
+#include "Engine/Engine.h"
+
+std::vector<AttackDescription> attackList;
+
+// Original function was replaced
+//----- (0040261D) --------------------------------------------------------
+// void stru298::Add(int16_t uID, int16_t a3, int16_t x, int16_t y, int16_t z, ABILITY_INDEX a7, char a8);
+
+void pushMeleeAttack(int16_t uID, Vec3i pos, ABILITY_INDEX ability) {
+    // Registered melee attack use the same distance as ranged attacks
+    attackList.push_back({uID, pos, engine->config->gameplay.RangedAttackDepth.Get(), true, ability});
```

Comment:
I think we shouldn't use config values for purposes that are not obvious from the config value name.

How about we just create a separate config value to use here?

Reply:
I can but I am not entirely sure about practical usage of this distance.
It is for melee so distance between attacker and it's target should be minimal.
Only one thing I can think of is that target of attack was teleported or something between frames where attack was registered and next frame where they were processed.

Reply:
Yeah, you're right that it's not very practical. OK let's leave RangedAttackDepth here.